### PR TITLE
KEP-4358: Custom resource field selectors GA

### PR DESCRIPTION
**Pull request by jpbetz**
KEP: ~https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/95-custom-resource-definitions/README.md~
https://github.com/kubernetes/enhancements/issues/4358

### CODE REVIEW

**Review by sftim — COMMENTED** [review]
Thanks. I think we should also revise https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/ because the feature gate is now locked to true (well, it will be for v1.32)

/hold
pending confirmation that the code change has gone in to [k/k](https://github.com/kubernetes/kubernetes)

**Comment by jpbetz** [comment]
> Thanks. I think we should also revise https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/ because the feature gate is now locked to true (well, it will be for v1.32)

Updated.

**Comment by tengqm** [comment]
I did a quick search but I failed to find the upstream source code change related to this promotion.

**Comment by jpbetz** [comment]
> I did a quick search but I failed to find the upstream source code change related to this promotion.

Thanks! I just opened https://github.com/kubernetes/kubernetes/pull/127673 but it might take time to merge because it's backed up on some feature gate stuff.

/hold
Until https://github.com/kubernetes/kubernetes/pull/127673 merges

**Comment by jpbetz** [comment]
/hold cancel
https://github.com/kubernetes/kubernetes/pull/127673 has merged

**Comment by jpbetz** [comment]
Friendly nudge that this is ready to merge

**Comment by jpbetz** [comment]
Friendly nudge on this

**Review by dipesh-rawat — COMMENTED** [review]
@jpbetz We might also want to make some revisions to the content in the **_‘Custom Resources’_** concept page under the "**Selectable fields for custom resources"** section.

- Reference link - [https://k8s.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#crd-selectable-fields](https://deploy-preview-48089--kubernetes-io-main-staging.netlify.app/docs/concepts/extend-kubernetes/api-extension/custom-resources/#crd-selectable-fields)

- Markdown file - [/docs/concepts/extend-kubernetes/api-extension/custom-resources.md](https://github.com/kubernetes/website/blob/dev-1.31/content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md)

**Comment by jpbetz** [comment]
> Joe Betz We might also want to make some revisions to the content in the **_‘Custom Resources’_** concept page under the "**Selectable fields for custom resources"** section.
> 
> * Reference link - [https://k8s.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#crd-selectable-fields](https://deploy-preview-48089--kubernetes-io-main-staging.netlify.app/docs/concepts/extend-kubernetes/api-extension/custom-resources/#crd-selectable-fields)
> * Markdown file - [/docs/concepts/extend-kubernetes/api-extension/custom-resources.md](https://github.com/kubernetes/website/blob/dev-1.31/content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md)

Thanks!  I added this section in Beta and nothing has changed since promoting to GA.  Do you see anything specific that needs my attention?

**Comment by sftim** [comment]
Check the wording of https://deploy-preview-48089--kubernetes-io-main-staging.netlify.app/docs/concepts/extend-kubernetes/api-extension/custom-resources/#crd-selectable-fields

We should drop the mention about whether you have enabled the feature gate; once we lock them to enabled, the point is settled.

But also, we should try to incorporate some of this text in the general page. It's no longer a feature that you can opt of of; it's just part of the CRD API.

---

Also, please update https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/ to link to the relevant part of the CRD docs.

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md [hunk 1]
```diff
@@ -1686,9 +1686,7 @@ may also be used with field selectors when included in the `spec.versions[*].sel
 {{< feature-state feature_gate_name="CustomResourceFieldSelectors" >}}
 
 For Kubernetes {{< skew currentVersion >}} the ability to define field selectors for
-custom resources is available by default (enabled by default since Kubernetes v1.31);
-you can disable it for your cluster  by turning off the `CustomResourceFieldSelectors`
-[feature gate](/docs/reference/command-line-tools-reference/feature-gates/).
+custom resources is a stable feature since Kubernetes v1.32.
```
Comment (sftim) [inline content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md:1689]:
```suggestion
```

The previous `feature-state` shortcode conveys the same thing.

**Review by sftim — COMMENTED** [review]
LGTM for docs. Does this need a tech review?

**Comment by sftim** [comment]
@jpbetz please link to the KEP from the PR description. Or if there is no KEP, to the promotion PR in [k/k](https://github.com/kubernetes/kubernetes/).

**Comment by jpbetz** [comment]
> LGTM for docs. Does this need a tech review?

I don't think so. All the substance of this documentation was tech reviewed for Beta.  I can't imagine what value I'd be asking a tech reviewer to add for these edits.

**Comment by sftim** [comment]
This is for https://github.com/kubernetes/enhancements/issues/4358

/lgtm

**Comment by sftim** [comment]
/remove-label tide/merge-method-squash

3 commits is fine (30 wouldn't be!)

**Pull request merged**